src/energyplus/ReverseTranslator/ReverseTranslateCoilCoolingDX.cpp

```diff
@@ -90,10 +92,14 @@ namespace energyplus {
     }
 
     if ((target = workspaceObject.getTarget(openstudio::Coil_Cooling_DXFields::CondenserZoneName))) {
-      OptionalModelObject modelObject = translateAndMapWorkspaceObject(*target);
-      if (modelObject) {
-        if (auto optZ = modelObject->optionalCast<ThermalZone>()) {
-          dx.setCondenserZone(optZ.get());
+      if (auto mo_ = translateAndMapWorkspaceObject(*target)) {
+        // Zone is translated, and a Space is returned instead
+        if (boost::optional<Space> space_ = mo_->optionalCast<Space>()) {
+          if (auto z_ = space_->thermalZone()) {
+            dx.setCondenserZone(z_.get());
+          }
+        } else {
+          LOG(Warn, workspaceObject.briefDescription() << " has a wrong type for 'Condenser Zone Name'");
```

Comment:
I found this same issue in a bunch of existing RT files so I fixed those I found.